### ISSUE TITLE
Android crash with java.lang.SecurityException

### ISSUE THREAD

**Issue by pitAlex** [phonegap/phonegap-plugin-push#2292]
Hi,
We are using the latest version of the push plugin and we noticed we had one crash reported in the 
play console which looks like this:

```
java.lang.RuntimeException: 
  at android.app.ActivityThread.performPauseActivityIfNeeded (ActivityThread.java:3897)
  at android.app.ActivityThread.performPauseActivity (ActivityThread.java:3856)
  at android.app.ActivityThread.performPauseActivity (ActivityThread.java:3830)
  at android.app.ActivityThread.handlePauseActivity (ActivityThread.java:3804)
  at android.app.ActivityThread.-wrap16 (ActivityThread.java)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1601)
  at android.os.Handler.dispatchMessage (Handler.java:110)
  at android.os.Looper.loop (Looper.java:203)
  at android.app.ActivityThread.main (ActivityThread.java:6309)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:1063)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:924)

Caused by: java.lang.SecurityException: 
  at android.os.Parcel.readException (Parcel.java:1683)
  at android.os.Parcel.readException (Parcel.java:1636)
  at android.app.INotificationManager$Stub$Proxy.cancelAllNotifications (INotificationManager.java:794)
  at android.app.NotificationManager.cancelAll (NotificationManager.java:374)

  at com.adobe.phonegap.push.PushPlugin.clearAllNotifications (PushPlugin.java:489)
  at com.adobe.phonegap.push.PushPlugin.onPause (PushPlugin.java:469)

  at org.apache.cordova.PluginManager.onPause (PluginManager.java:211)
  at org.apache.cordova.CordovaWebViewImpl.handlePause (CordovaWebViewImpl.java:433)
  at org.apache.cordova.CordovaActivity.onPause (CordovaActivity.java:245)
  at android.app.Activity.performPause (Activity.java:6873)
  at android.app.Instrumentation.callActivityOnPause (Instrumentation.java:1322)
  at android.app.ActivityThread.performPauseActivityIfNeeded (ActivityThread.java:3879)
```
This is for [TCT (Alcatel) 5049W (Mickey6TTMO)](https://www.handsetdetection.com/device-detection-database/devices/alcatel/5049w/) with an Android 7.0.  There seems to be some security violated when the app is paused and notifications are being cleared.

**Comment by macdonst** [comment]
Sorry to be a jerk but I'm now closing issues that don't follow the issue submission guidelines. This plugin takes a lot of work to maintain and I can't afford the time required to solicit the required info from everyone.

I'm closing this issue but please feel free to re-open the issue if you provide the requested details from the new issue submission guidelines.

https://github.com/phonegap/phonegap-plugin-push/blob/master/.github/ISSUE_TEMPLATE.md

**Comment by pitAlex** [comment]
Excuse me?? Are you reading what I posted or just decided to come up with the "issue submission guidelines"  shit because you don't have any idea whats going on here?

This is the live app (crash reported in the play console )  - so no way to replicate, no way to know what the behavior is. The stack-trace is all there is.

Now instead of being a jerk you could've at least provide info on what the code does there, what api is it accessing and such.
I can't believe you would even try an pull that with me when there are other issues reported which don't follow the template either.

**Comment by fredgalvao** [comment]
You did **not** inform any of the following information, which is very important to even begin to investigate anything, and are explicit on the issue template:
- cordova version associated
- cordova-android version associated
- phonegap-plugin-push version associated
- a single line of code showing your usage of this plugin

If it's in your interest to solve whatever issue your users are having, I'd suggest you try to be friendlier and partially follow as many sections as possible from the issue template. I assure you there are a bunch of people here willing  to help if you're willing to help too.

**Comment by pitAlex** [comment]
Its not always needed. And I did say I am using the latest version of the
plugin. You can assume I am using the latest of cordova and the
cordova-android.
As for code, there is no code to provide when this is happening at the
native level, automatically. If you check the stacktrace it fails right
here:

SharedPreferences prefs = getApplicationContext().getSharedPreferences(
COM_ADOBE_PHONEGAP_PUSH,
Context.MODE_PRIVATE);
if (prefs.getBoolean(CLEAR_NOTIFICATIONS, true)) {
clearAllNotifications(); // < - here
} In PushPlugin.java line 469.
What code sample is to provide in this situation?

Anyway, I've already taken care of it on my side. I suggest you add a try
catch code there, it seems to be some system issue and not something
related to the push code. It makes sense considering it happened only once
so far.

**Comment by jcesarmobile** [comment]
No, we can't assume anything, there are a ton of people using latest version of the plugin on previous versions of cordova or cordova-android, and people saying they are using latest, but not really using latest, or if somebody looks into your issue in a few days latest might not be your current version because of an update and we can't go and check which version was latest today.

We also can't know if you are calling init with `clearBadge` option or if you are calling`clearAllNotifications` method yourself or if it's automatically called just by a stacktrace.

So please, provide all the information required and be nicer.
